src/MongoDB.Bson/Serialization/BsonClassMap.cs

```diff
@@ -426,6 +421,42 @@ public static void RegisterClassMap(BsonClassMap classMap)
             }
         }
 
+        // private static methods
+        private static Type GetFormatterServicesType()
+        {
+#if NET452
+            return typeof(FormatterServices);
+#else
+            // TODO: once we depend on newer versions of .NET Standard we should be able to do this without reflection
+
+            Type formatterServicesType;
+            try
+            {
+                // new approach which works on .NET Core 3.0
+                var formattersAssembly = Assembly.Load(new AssemblyName("System.Runtime.Serialization.Formatters"));
+                formatterServicesType = formattersAssembly.GetType("System.Runtime.Serialization.FormatterServices");
+                if (formatterServicesType != null)
+                {
+                    return formatterServicesType;
+                }
+            }
+            catch
+            {
+                // ignore exceptions and continue to fallback code
+            }
+
+            // fallback to previous approach (which worked on older versions of .NET Core)
```

Comment:
The fallback is required in order to continue to work on .NET Core 1.1 (tested with a Console app).

The new approach only works on newer versions of .NET Core (I tested against 2.1).